packages/git-resolver/src/parsePref.ts

```diff
@@ -30,6 +30,9 @@ const gitProtocols = new Set([
 ])
 
 export default async function parsePref (pref: string): Promise<HostedPackageSpec | null> {
+  if (pref.startsWith('https://github.com/') || pref.startsWith('http://github.com/')) {
```

Comment:
If gitlab and bitbucket would work the same way:

```suggestion
  if (/^http?://(git(hub|lab)|bitbucket)\.com/.test(pref)) {
```

Reply:
I think gitlab and bitbucket would need to have the package path prefixed with `gitlab:` and `bitbucket:` (e.g. `gitlab:pnpm/pnpm`). IIUC, the `pref.replace(..., '')` works for github because github is special cased to be the default git provider.